### PR TITLE
DFC-65 / DFC-67 / DFC-70 add pageview tracker and data push functions

### PR DESCRIPTION
**Pull request by di-fabs**
### Description ###
- Create baseTracker module
- Add trackOnPageLoad function
- Update the package documentation
- add pageViewTracker object into analyticsGa4

### Tickets ###
(DFC-65)[https://govukverify.atlassian.net/jira/software/c/projects/DFC/boards/436?selectedIssue=DFC-65]
(DFC-67)[https://govukverify.atlassian.net/jira/software/c/projects/DFC/boards/436?selectedIssue=DFC-67]
(DFC-70)[https://govukverify.atlassian.net/jira/software/c/projects/DFC/boards/436?selectedIssue=DFC-70]

### CODE REVIEW

File: PACKAGE-README.md [hunk 1]
```diff
@@ -80,4 +80,27 @@ This package enables GOV UK LOGIN frontend Node.js applications to use Google Ta
    ```
    [!NOTE] window.DI.appInit is a function loaded from analytics.js. That will create a new instance of our analytics library and store into window.DI.analyticsGa4 [!NOTE]
 
+### Page View Tracker
+
+Page view tracking allows us to see which pages are most visited, where your visitors are coming from, etc.
+It can be called by using the method trackOnPageLoad of the object pageViewTracker stored into the analytics library (analyticsGa4)
+
+It takes as unique parameter an object define by :
```
Comment (SophiaDWokoma) [inline PACKAGE-README.md:88]: A typo I think , it should be  "It takes as **a** unique ..."

File: PACKAGE-README.md [hunk 2]
```diff
@@ -80,4 +80,27 @@ This package enables GOV UK LOGIN frontend Node.js applications to use Google Ta
...
+- statusCode: number
+- englishPageTitle: string
+- taxonomy_level1: string
+- taxonomy_level2: string
+
+Example:
+
+```js
+window.DI.analyticsGa4.pageViewTracker.trackOnPageLoad({
+  statsCode: 200,
```
Comment (SophiaDWokoma) [inline PACKAGE-README.md:99]: Another typo - **status**

File: PACKAGE-README.md [hunk 3]
```diff
@@ -80,4 +80,27 @@ This package enables GOV UK LOGIN frontend Node.js applications to use Google Ta
...
+  englishPageTitle: "english version of the page title",
+  taxonomy_level1: "test tax1",
+  taxotaxonomy_level2: "test tax2",
```
Comment (SophiaDWokoma) [inline PACKAGE-README.md:102]: Another typo - **taxonomy**

File: src/analytics/baseTracker/baseTracker.test.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    expect(languageCode).toEqual(document.documentElement.lang);
+  });
+
+  test("Get en as a default Language", () => {
+    document.documentElement.lang = "";
+    const languageCode = newInstance.getLanguage();
+    expect(languageCode).toEqual("en");
+  });
+
+  test("Get Language from html tag", () => {
```
Comment (SophiaDWokoma) [inline src/analytics/baseTracker/baseTracker.test.ts:43]: Is this test case a duplicate of the first one on line 31 ?

Comment (di-fabs) [reply src/analytics/baseTracker/baseTracker.test.ts:43]: Well spotted!

**Review by di-aholme — APPROVED** [review]

**Review by SophiaDWokoma — APPROVED** [review]

**Pull request merged**